sycl/source/detail/image_impl.cpp

```diff
@@ -76,6 +76,158 @@ uint8_t getImageElementSize(uint8_t NumChannels, image_channel_type Type) {
   return Retval;
 }
 
+RT::PiChannelOrder convertChannelOrder(image_channel_order Order) {
+  switch (Order) {
+  case image_channel_order::a:
+    return PI_A;
+  case image_channel_order::r:
+    return PI_R;
+  case image_channel_order::rx:
+    return PI_Rx;
+  case image_channel_order::rg:
+    return PI_RG;
+  case image_channel_order::rgx:
+    return PI_RGx;
+  case image_channel_order::ra:
+    return PI_RA;
+  case image_channel_order::rgb:
+    return PI_RGB;
+  case image_channel_order::rgbx:
+    return PI_RGBx;
+  case image_channel_order::rgba:
+    return PI_RGBA;
+  case image_channel_order::argb:
+    return PI_ARGB;
+  case image_channel_order::bgra:
+    return PI_BGRA;
+  case image_channel_order::intensity:
+    return PI_INTENSITY;
+  case image_channel_order::luminance:
+    return PI_LUMINANCE;
+  case image_channel_order::abgr:
+    return PI_ABGR;
+  default: {
+    assert(!"Unhandled image_channel_order");
+    return static_cast<RT::PiChannelOrder>(0);
+  }
+  }
+}
+
+image_channel_order convertChannelOrder(RT::PiChannelOrder Order) {
+  switch (Order) {
+  case PI_A:
+    return image_channel_order::a;
+  case PI_R:
+    return image_channel_order::r;
+  case PI_Rx:
+    return image_channel_order::rx;
+  case PI_RG:
+    return image_channel_order::rg;
+  case PI_RGx:
+    return image_channel_order::rgx;
+  case PI_RA:
+    return image_channel_order::ra;
+  case PI_RGB:
+    return image_channel_order::rgb;
+  case PI_RGBx:
+    return image_channel_order::rgbx;
+  case PI_RGBA:
+    return image_channel_order::rgba;
+  case PI_ARGB:
+    return image_channel_order::argb;
+  case PI_BGRA:
+    return image_channel_order::bgra;
+  case PI_INTENSITY:
+    return image_channel_order::intensity;
+  case PI_LUMINANCE:
+    return image_channel_order::luminance;
+  case PI_ABGR:
+    return image_channel_order::abgr;
+  default: {
+    assert(!"Unhandled image_channel_order");
+    return static_cast<image_channel_order>(0);
+  }
+  }
```

Comment:
I think conversion from OpenCL values to PI values should be done by utilities defined in OpenCL plug-in rather than here.

Reply:
These functions convert  from sycl values to PI ones.